webapp/src/features/adminAdduser/adminAddUserSlice.js

```diff
@@ -0,0 +1,204 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { selectToken, selectOrganizationName } from "../../generalSlices/userSlice";
+import EnvironmentVars from "../../EnvironmentVars";
+import apiHelper from "../../apis/api-helper";
+import { getAvailableUsers } from "../adminUserTab/adminUserTabSlice";
+
+const initialState = {
+  successMsg: "",
+  selectedOrganizationNames: [],
+  selectedOrganizations: [],
+  organizationNames: [],
+  availableRoles: [],
+  apiData: {},
+  errorState: false,
+  errorMsg: "",
+  submitAttempt: false,
+};
+
+export const getUserData = createAsyncThunk(
+  "adminAddUser/getUserData",
+  async (_, { getState }) => {
+    const currentState = getState();
+    const token = selectToken(currentState);
+
+    const data = await apiHelper._getDataWithCodes({
+      url: EnvironmentVars.adminAddUser,
+      token,
+      additional_headers: { "Content-Type": "application/json" },
+    });
+
+    switch (data.status) {
+      case 200:
+        return { success: true, message: "", data: data.body };
+      case 400:
+      case 500:
+        return { success: false, message: data.message };
+      default:
+        return;
```

Comment:
In the default case should we still return an object with success false, and maybe an empty message or something?

Reply:
updated to match the 400/500 case